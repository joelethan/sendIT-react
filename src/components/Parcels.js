import React from 'react'

export default function Parcels(props) {

   const data={props};
    
  return (
    <tr>
    <td>{data.pickup_location}</td>
    </tr>
  )
}
