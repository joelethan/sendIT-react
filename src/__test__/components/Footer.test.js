import React from "react";
import { shallow } from "enzyme";
import Footer from "../../components/Footer";
import Parcels from "../../components/Parcels";
import Login from "../../components/Login";
import CreateParcels from "../../components/CreateParcels";
import Landing from "../../components/Landing";
import Register from "../../components/Register";
import Navigation from "../../components/Navigation";



it("renders without crashing", () => {
  const wrapper = shallow(<CreateParcels />);
  
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Footer />);
  
  expect(wrapper).toMatchSnapshot();
});



it("renders without crashing", () => {
  const wrapper = shallow(<Landing />);
  
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Login />);
  
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Parcels />);
  
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Login />);
  
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Register />);
  
  expect(wrapper).toMatchSnapshot();
});

it("renders without crashing", () => {
  const wrapper = shallow(<Navigation />);
  
  expect(wrapper).toMatchSnapshot();
});
