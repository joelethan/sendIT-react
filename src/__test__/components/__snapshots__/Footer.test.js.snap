// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(CreateParcels) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders without crashing 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Copyright © ",
        2019,
        " SentIT",
      ],
      "className": "bg-dark text-white mt-5 p-4 text-center ",
    },
    "ref": null,
    "rendered": Array [
      "Copyright © ",
      2019,
      " SentIT",
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Copyright © ",
          2019,
          " SentIT",
        ],
        "className": "bg-dark text-white mt-5 p-4 text-center ",
      },
      "ref": null,
      "rendered": Array [
        "Copyright © ",
        2019,
        " SentIT",
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="landing"
      >
        <div
          className="dark-overlay landing-inner text-light"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 text-center"
              >
                <h1
                  className="display-3 mb-4"
                >
                  SendIT
                </h1>
                <p
                  className="lead"
                >
                   Contact us today to have your parcels delivered at the destination safely and on time
                </p>
                <hr />
                <Link
                  className="btn btn-lg btn-info mr-2"
                  replace={false}
                  to="/register"
                >
                  Sign Up
                </Link>
                <Link
                  className="btn btn-lg btn-light"
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="dark-overlay landing-inner text-light"
        >
          <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 text-center"
              >
                <h1
                  className="display-3 mb-4"
                >
                  SendIT
                </h1>
                <p
                  className="lead"
                >
                   Contact us today to have your parcels delivered at the destination safely and on time
                </p>
                <hr />
                <Link
                  className="btn btn-lg btn-info mr-2"
                  replace={false}
                  to="/register"
                >
                  Sign Up
                </Link>
                <Link
                  className="btn btn-lg btn-light"
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </div>
            </div>
          </div>
        </div>,
        "className": "landing",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
          >
            <div
              className="row"
            >
              <div
                className="col-md-12 text-center"
              >
                <h1
                  className="display-3 mb-4"
                >
                  SendIT
                </h1>
                <p
                  className="lead"
                >
                   Contact us today to have your parcels delivered at the destination safely and on time
                </p>
                <hr />
                <Link
                  className="btn btn-lg btn-info mr-2"
                  replace={false}
                  to="/register"
                >
                  Sign Up
                </Link>
                <Link
                  className="btn btn-lg btn-light"
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </div>
            </div>
          </div>,
          "className": "dark-overlay landing-inner text-light",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="row"
            >
              <div
                className="col-md-12 text-center"
              >
                <h1
                  className="display-3 mb-4"
                >
                  SendIT
                </h1>
                <p
                  className="lead"
                >
                   Contact us today to have your parcels delivered at the destination safely and on time
                </p>
                <hr />
                <Link
                  className="btn btn-lg btn-info mr-2"
                  replace={false}
                  to="/register"
                >
                  Sign Up
                </Link>
                <Link
                  className="btn btn-lg btn-light"
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </div>
            </div>,
            "className": "container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="col-md-12 text-center"
              >
                <h1
                  className="display-3 mb-4"
                >
                  SendIT
                </h1>
                <p
                  className="lead"
                >
                   Contact us today to have your parcels delivered at the destination safely and on time
                </p>
                <hr />
                <Link
                  className="btn btn-lg btn-info mr-2"
                  replace={false}
                  to="/register"
                >
                  Sign Up
                </Link>
                <Link
                  className="btn btn-lg btn-light"
                  replace={false}
                  to="/login"
                >
                  Login
                </Link>
              </div>,
              "className": "row",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="display-3 mb-4"
                  >
                    SendIT
                  </h1>,
                  <p
                    className="lead"
                  >
                     Contact us today to have your parcels delivered at the destination safely and on time
                  </p>,
                  <hr />,
                  <Link
                    className="btn btn-lg btn-info mr-2"
                    replace={false}
                    to="/register"
                  >
                    Sign Up
                  </Link>,
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>,
                ],
                "className": "col-md-12 text-center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SendIT",
                    "className": "display-3 mb-4",
                  },
                  "ref": null,
                  "rendered": "SendIT",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": " Contact us today to have your parcels delivered at the destination safely and on time",
                    "className": "lead",
                  },
                  "ref": null,
                  "rendered": " Contact us today to have your parcels delivered at the destination safely and on time",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Sign Up",
                    "className": "btn btn-lg btn-info mr-2",
                    "replace": false,
                    "to": "/register",
                  },
                  "ref": null,
                  "rendered": "Sign Up",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Login",
                    "className": "btn btn-lg btn-light",
                    "replace": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="landing"
        >
          <div
            className="dark-overlay landing-inner text-light"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-12 text-center"
                >
                  <h1
                    className="display-3 mb-4"
                  >
                    SendIT
                  </h1>
                  <p
                    className="lead"
                  >
                     Contact us today to have your parcels delivered at the destination safely and on time
                  </p>
                  <hr />
                  <Link
                    className="btn btn-lg btn-info mr-2"
                    replace={false}
                    to="/register"
                  >
                    Sign Up
                  </Link>
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>
                </div>
              </div>
            </div>
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="dark-overlay landing-inner text-light"
          >
            <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-12 text-center"
                >
                  <h1
                    className="display-3 mb-4"
                  >
                    SendIT
                  </h1>
                  <p
                    className="lead"
                  >
                     Contact us today to have your parcels delivered at the destination safely and on time
                  </p>
                  <hr />
                  <Link
                    className="btn btn-lg btn-info mr-2"
                    replace={false}
                    to="/register"
                  >
                    Sign Up
                  </Link>
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>
                </div>
              </div>
            </div>
          </div>,
          "className": "landing",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
            >
              <div
                className="row"
              >
                <div
                  className="col-md-12 text-center"
                >
                  <h1
                    className="display-3 mb-4"
                  >
                    SendIT
                  </h1>
                  <p
                    className="lead"
                  >
                     Contact us today to have your parcels delivered at the destination safely and on time
                  </p>
                  <hr />
                  <Link
                    className="btn btn-lg btn-info mr-2"
                    replace={false}
                    to="/register"
                  >
                    Sign Up
                  </Link>
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>
                </div>
              </div>
            </div>,
            "className": "dark-overlay landing-inner text-light",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="row"
              >
                <div
                  className="col-md-12 text-center"
                >
                  <h1
                    className="display-3 mb-4"
                  >
                    SendIT
                  </h1>
                  <p
                    className="lead"
                  >
                     Contact us today to have your parcels delivered at the destination safely and on time
                  </p>
                  <hr />
                  <Link
                    className="btn btn-lg btn-info mr-2"
                    replace={false}
                    to="/register"
                  >
                    Sign Up
                  </Link>
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>
                </div>
              </div>,
              "className": "container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="col-md-12 text-center"
                >
                  <h1
                    className="display-3 mb-4"
                  >
                    SendIT
                  </h1>
                  <p
                    className="lead"
                  >
                     Contact us today to have your parcels delivered at the destination safely and on time
                  </p>
                  <hr />
                  <Link
                    className="btn btn-lg btn-info mr-2"
                    replace={false}
                    to="/register"
                  >
                    Sign Up
                  </Link>
                  <Link
                    className="btn btn-lg btn-light"
                    replace={false}
                    to="/login"
                  >
                    Login
                  </Link>
                </div>,
                "className": "row",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="display-3 mb-4"
                    >
                      SendIT
                    </h1>,
                    <p
                      className="lead"
                    >
                       Contact us today to have your parcels delivered at the destination safely and on time
                    </p>,
                    <hr />,
                    <Link
                      className="btn btn-lg btn-info mr-2"
                      replace={false}
                      to="/register"
                    >
                      Sign Up
                    </Link>,
                    <Link
                      className="btn btn-lg btn-light"
                      replace={false}
                      to="/login"
                    >
                      Login
                    </Link>,
                  ],
                  "className": "col-md-12 text-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "SendIT",
                      "className": "display-3 mb-4",
                    },
                    "ref": null,
                    "rendered": "SendIT",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": " Contact us today to have your parcels delivered at the destination safely and on time",
                      "className": "lead",
                    },
                    "ref": null,
                    "rendered": " Contact us today to have your parcels delivered at the destination safely and on time",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Sign Up",
                      "className": "btn btn-lg btn-info mr-2",
                      "replace": false,
                      "to": "/register",
                    },
                    "ref": null,
                    "rendered": "Sign Up",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Login",
                      "className": "btn btn-lg btn-light",
                      "replace": false,
                      "to": "/login",
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": "div",
            },
            "type": "div",
          },
          "type": "div",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Login) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders without crashing 5`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Parcels />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <td />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "td",
    },
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <td />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "td",
      },
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`renders without crashing 6`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Login) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders without crashing 7`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(withRouter(Register)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`renders without crashing 8`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Navigation) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "_calculateChangedBits": null,
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "_calculateChangedBits": null,
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": Object {
            "$$typeof": Symbol(react.provider),
            "_context": [Circular],
          },
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
